### PR TITLE
Removed Extra condition for checking expr is zero or not

### PR DESCRIPTION
**Pull request by abhaydhiman**
#### References to other Issues or PRs

Nan

#### Brief description of what is fixed or changed
In **simplify** of **simplify.py**, there is an if statement `if isinstance(expr, Expr) and expr.is_zero and expr*zero == S.zero` with an extra condition `expr*zero == S.zero` that always gives **True** bool value until the expression is

 undefined. But that can also be done only by `expr.is_zero`.

#### Other comments

```
 expr_list = [
    x^3 - x^3,
    x^3 - x^3 + x,
    x^3 - x^3,
    (x^3 - x^3 + x)/ (x^3 - x^3),
    (x^3 - x^3)/ (x^3 - x^3 + x),
    sp.simplify((sp.sin(x)^2 - x^3 + x)/ (sp.sin(x)^2 + sp.cos(x)^2)),
    sp.simplify((sp.sin(x)^2 - 1 + sp.cos(x)^2)/ (sp.sin(x)^2 + sp.cos(x)^2))
     ]
def check_conditions(expr):
    print('Expression-->', expr)
    print('checking is_instance with Expr-->', isinstance(expr, Expr))
    print('checking is_zero on the expression-->', expr.is_zero)
    print('checking final condition-->', expr*0 == S.Zero)
    print()
for exprs in expr_list:
    check_conditions(exprs)
---------------------------------------------------------------------------------------------------------------------
Expression--> 0
checking is_instance with Expr--> True
checking is_zero on the expression--> True
checking final condition--> True

Expression--> x
checking is_instance with Expr--> True
checking is_zero on the expression--> None
checking final condition--> True

Expression--> 0
checking is_instance with Expr--> True
checking is_zero on the expression--> True
checking final condition--> True

Expression--> zoo*x
checking is_instance with Expr--> True
checking is_zero on the expression--> None
checking final condition--> False

Expression--> 0
checking is_instance with Expr--> True
checking is_zero on the expression--> True
checking final condition--> True

Expression--> -x**3 + x + sin(x)**2
checking is_instance with Expr--> True
checking is_zero on the expression--> None
checking final condition--> True

Expression--> 0
checking is_instance with Expr--> True
checking is_zero on the expression--> True
checking final condition--> True
```

#### Release Notes

NO ENTRY

### CODE REVIEW

**Comment by oscarbenjamin** [comment]
I presume that was added for a reason at the time. Can you track down the commit and pull request where it was added using blame?

**Comment by abhaydhiman** [comment]
> I presume that was added for a reason at the time. Can you track down the commit and pull request where it was added using blame?

No, I didn't track down the commit or the pull request. I am just going through the code and found this so I committed. I also don't know how to use blame can you give me some kind of referencing to do this.

**Comment by oscarbenjamin** [comment]
The easiest way to see the blame is on github. You can got to the sympy repo and then navigate to the file and then click the blame button. That will show you which commit each line was last changed in. You can click on the commit and then that will show which PR it was merged in. From the blame you can also click to see the blame from before any particular commit.

**Comment by abhaydhiman** [comment]
These are the commits for the particular code change in this PR
> 7649a7c
ea0540f

but now only `expr.is_zero` is working properly for the cases.

**Comment by abhaydhiman** [comment]
```
import sympy as sp

def check_conditions(expr):
    print('Output:-')
    print('Expression-->', expr)
    print('Type of Expression-->', type(expr))
    print('checking is_instance with Expr-->', isinstance(expr, Expr))
    print('checking is_zero on the expression-->', expr.is_zero)
    print('checking final condition-->', expr*0 == S.Zero)
    print()

expr = sp.Integral(x, (x, 1, 1))
check_conditions(expr)
-------------------------------------------------------------------------------------------------------------------
Output:-
Expression--> Integral(x, (x, 1, 1))
Type of Expression--> <class 'sympy.integrals.integrals.Integral'>
checking is_instance with Expr--> True
checking is_zero on the expression--> True
checking final condition--> True
```


from the above code, it is clear that expr.is_zero is sufficient to check the expr IS_ZERO

**Comment by oscarbenjamin** [comment]
I think I understand the reason for this. Previously a zero matrix would have had `is_zero = True`. The test `expr*0 is S.Zero` is meant to exclude that case. This is now fixed I think so that `is_zero` is only true for things that are equal to the number zero.

**Comment by abhaydhiman** [comment]
But this `isinstance(expr, Expr)` would become `False` if a zero matrix is passed, I guess. Ultimately there is no need for `expr*zero == S.zero`

**Comment by abhaydhiman** [comment]
If I am wrong please correct me.

**Comment by oscarbenjamin** [comment]
A standard mutable Matrix is not an Expr subclass but ImmutableMatrix or other matrix classes are. For example in sympy 1.5.1:
```julia
In [1]: z = ImmutableMatrix([[0, 0]])                                                                                                          

In [2]: z                                                                                                                                      
Out[2]: [0  0]

In [3]: z.is_zero                                                                                                                              
Out[3]: True

In [4]: isinstance(z, Expr)                                                                                                                    
Out[4]: True
```
That's fixed now on master and in 1.6:
```julia
In [1]: z = ImmutableMatrix([[0, 0]])                                                                                                          

In [4]: z.is_zero_matrix                                                                                                                       
Out[4]: True
```
It should be the case now that any Expr subclass having `is_zero=True` should represent the actual number zero rather than something else.

File: sympy/concrete/tests/test_sums_products.py [hunk 1]
```diff
@@ -1053,6 +1053,10 @@ def test_is_convergent():
     # issue 19545
     assert Sum(1/n - 3/(3*n +2), (n, 1, oo)).is_convergent() is S.true
 
+    # issue 19836
+    assert Sum(4/(n + 2) - 5/(n + 1) + 1/n,(n,7,oo)).is_convergent() is S.true
```
Comment (oscarbenjamin) [inline sympy/concrete/tests/test_sums_products.py:1057]: Is this test related to the other change (about `is_zero`)?

**Comment by oscarbenjamin** [comment]
This now has a merge conflict because the branch here includes the change from your other PR

File: sympy/concrete/tests/test_sums_products.py (hunk 1 above)
Comment (abhaydhiman) [reply sympy/concrete/tests/test_sums_products.py:1057]: Actually I am new to this and I made a mess that I know. So can I make another PR that just removing `expr*0 == S.Zero` from the code?

Comment (oscarbenjamin) [reply sympy/concrete/tests/test_sums_products.py]: It's probably best to do that. Otherwise you can reuse this PR with:
```
$ git reset HEAD^
< check git diff, undo the unrelated change to the code >
$ git add <filename>
$ git commit
$ git push --force
```

Comment (abhaydhiman) [reply sympy/concrete/tests/test_sums_products.py]: I have done the changes in this PR.

**Comment by oscarbenjamin** [comment]
Okay, looks good. Thanks

**Comment by abhaydhiman** [comment]
> Okay, looks good. Thanks

In fact, I am thankful to you as you guided me a lot.
